mint-client/src/clients/user.rs

```diff
@@ -379,6 +388,49 @@ impl UserClient {
             .await
             .map_err(|_| ClientError::WaitContractTimeout)?
     }
+
+    pub fn create_invoice<R: RngCore + CryptoRng>(
+        &self,
+        amount: Amount,
+        mut rng: R,
+    ) -> Result<Invoice, ClientError> {
+        // Hard-coding preimage (pubkey) for now
+        let raw_payment_secret = [0; 32];
+        let payment_hash = bitcoin::secp256k1::hashes::sha256::Hash::hash(&raw_payment_secret);
+        let payment_secret = PaymentSecret(raw_payment_secret);
+
+        // Final route to the user's contract inside the federation
+        let (node_secret_key, node_public_key) = self.context.secp.generate_keypair(&mut rng);
+
+        // Route hint instructing payer how to route to gateway
+        let gateway_route_hint = RouteHint(vec![RouteHintHop {
+            src_node_id: self.context.config.gateway.node_pub_key,
+            short_channel_id: 8,
+            fees: RoutingFees {
+                base_msat: 0,
+                proportional_millionths: 0,
+            },
+            cltv_expiry_delta: (8 << 4) | 1,
+            htlc_minimum_msat: None,
+            htlc_maximum_msat: None,
+        }]);
+
+        let invoice = InvoiceBuilder::new(Currency::Regtest)
+            .amount_milli_satoshis(amount.milli_sat)
+            .description("description".into())
+            .payment_hash(payment_hash)
+            .payment_secret(payment_secret)
+            .current_timestamp()
+            .min_final_cltv_expiry(144)
+            .payee_pub_key(node_public_key)
+            .private_route(gateway_route_hint)
+            .build_signed(|hash| self.context.secp.sign_recoverable(hash, &node_secret_key))?;
```

Comment:
Some of these parameters I just copied from examples in the docs. Don't know exactly what all of them mean ...